src/logging.cpp

```diff
@@ -224,10 +224,32 @@ std::string BCLog::Logger::LogTimestampStr(const std::string& str)
     return strStamped;
 }
 
+namespace {
+    /** Belts and suspenders: make sure outgoing log messages don't contain
+     * potentially suspicious characters, such as terminal control codes.
+     *
+     * This escapes control characters except newline ('\n') in C syntax.
+     * It escapes instead of removes them to still allow for troubleshooting
+     * issues where they accidentally end up in strings.
+     */
+    std::string LogEscapeMessage(const std::string& str) {
+        std::string ret;
+        for (char ch_in: str) {
+            uint8_t ch = (uint8_t)ch_in;
+            if (ch >= 32 || ch == '\n') {
+                ret += ch_in;
+            } else {
+                ret += strprintf("\\x%02x", ch);
```

Comment:
It seems impossible to distinguish between a literal "\\x00" and "\x00" after this transformation.

Suggest using `SanitizeString` perhaps with a new rule.

Reply:
I don't think that matters much.

Reply:
I'm very much against using SanitizeString's methodology here. I don't want to make any characters entirely disappear.

Reply:
Cherry-pick 7cbc72b93740a5c7e14d1fb1a9f45fd78ff24700?

Reply:
The escaping code there always escapes '%'